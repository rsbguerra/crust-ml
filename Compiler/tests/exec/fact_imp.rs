fn fact_imp(mut n: i32) -> i32 {
    let mut res = 1;
    while n > 1 {
        res = res * n;
        n = n - 1;
    }
    res
}

fn main() {
  if fact_imp(0) == 1 { print!("fact(0) = 1\n") }
  if fact_imp(1) == 1 { print!("fact(1) = 1\n"); }
  if fact_imp(5) == 120 { print!("fact(5) = 120\n"); }
  if fact_imp(12) == 479001600  { print!("fact(12) = 479001600\n"); }
}
