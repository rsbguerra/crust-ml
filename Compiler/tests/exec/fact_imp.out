fact(0) = 1
fact(1) = 1
fact(5) = 120
fact(12) = 479001600
